ibm_cloud_sdk_core/token_managers/container_token_manager.py

```diff
@@ -0,0 +1,163 @@
+            a "basic auth" Authorization header for interactions with the IAM token server.
+            Defaults to None.
+        client_secret: The client_id and client_secret fields are used to form
+            a "basic auth" Authorization header for interactions with the IAM token server.
+            Defaults to None.
+        disable_ssl_verification: A flag that indicates whether verification of
+            the server's SSL certificate should be disabled or not. Defaults to False.
+        headers: Default headers to be sent with every IAM token request. Defaults to None.
+        proxies: Proxies to use for communicating with IAM. Defaults to None.
+        proxies.http: The proxy endpoint to use for HTTP requests.
+        proxies.https: The proxy endpoint to use for HTTPS requests.
+        scope: The "scope" to use when fetching the bearer token from the IAM token server.
+        This can be used to obtain an access token with a specific scope.
+    """
+    DEFAULT_CR_TOKEN_FILENAME = '/var/run/secrets/tokens/vault-token'
+
+    def __init__(self,
+                 cr_token_filename: Optional[str] = None,
+                 iam_profile_name: Optional[str] = None,
+                 iam_profile_id: Optional[str] = None,
+                 url: Optional[str] = None,
+                 client_id: Optional[str] = None,
+                 client_secret: Optional[str] = None,
+                 disable_ssl_verification: bool = False,
+                 scope: Optional[str] = None,
+                 proxies: Optional[Dict[str, str]] = None,
+                 headers: Optional[Dict[str, str]] = None) -> None:
+        super().__init__(
+            url=url, client_id=client_id, client_secret=client_secret,
+            disable_ssl_verification=disable_ssl_verification, headers=headers, proxies=proxies, scope=scope)
+
+        self.cr_token_filename = cr_token_filename
+        self.iam_profile_name = iam_profile_name
+        self.iam_profile_id = iam_profile_id
+
+        self.request_payload['grant_type'] = 'urn:ibm:params:oauth:grant-type:cr-token'
+
+    def retrieve_cr_token(self) -> str:
+        """Retrieves the CR token for the current compute resource by reading it from the local file system.
+
+        Raises:
+            Exception: Cannot retrieve the compute resource token from.
+
+        Returns:
+            A string which contains the compute resource token.
+        """
+        cr_token_filename = self.cr_token_filename if self.cr_token_filename else self.DEFAULT_CR_TOKEN_FILENAME
+
+        logging.debug('Attempting to read CR token from file: %s', cr_token_filename)
+
+        try:
+            with open(cr_token_filename, 'r') as file:
+                cr_token = file.read()
+            return cr_token
+        # pylint: disable=broad-except
+        except Exception as ex:
+            raise Exception(
+                'Unable to retrieve the CR token value from file {}: {}'.format(cr_token_filename, ex)) from None
```

Comment:
I want to handle every error here, not just e.g. missing files, this is why I use the plain `Exception` here. If you think this is too common, I am open for suggestions.

Reply:
I think this is fine the way it is.  Not sure how much value there would be to creating a specific exception here (I assume that is what you were alluding to).